From e589847e74e823d47f3074bddb792bc43b62bfba Mon Sep 17 00:00:00 2001
From: Vladimir Zhbanov <vzhbanov@gmail.com>
Date: Thu, 18 Jun 2015 09:49:08 +0300
Subject: [PATCH 9/9] gaf: More portable invoking the export and config
 functions.

This fix targets MinGW port issues, that is, fixes commands 'gaf export'
and 'gaf config' which do not work in the MinGW environment due to
non-portable invoking of guile via scm_init_guile().
---
 gaf/config.c | 10 ++++++++--
 gaf/export.c | 13 ++++++++++---
 2 files changed, 18 insertions(+), 5 deletions(-)

diff --git a/gaf/config.c b/gaf/config.c
index 75d8ca6..935b8ae 100644
--- a/gaf/config.c
+++ b/gaf/config.c
@@ -68,8 +68,8 @@ config_usage (void)
 #define see_help_msg _("\nRun `gaf config --help' for more information.\n")
 #define multi_store_msg _("ERROR: You may only specify a single configuration store.\n")
 
-int
-cmd_config (int argc, char **argv)
+static void
+cmd_config_impl (void *data, int argc, char **argv)
 {
   int c;
   EdaConfig *cfg = NULL, *parent;
@@ -200,6 +200,12 @@ cmd_config (int argc, char **argv)
   }
 
   g_assert_not_reached ();
+}
 
+/* Main function for `gaf config' */
+int
+cmd_config (int argc, char **argv)
+{
+  scm_boot_guile (argc, argv, cmd_config_impl, NULL); /* Doesn't return */
   return 0;
 }
diff --git a/gaf/export.c b/gaf/export.c
index 5de0680..27319fc 100644
--- a/gaf/export.c
+++ b/gaf/export.c
@@ -146,9 +146,8 @@ static struct ExportSettings settings = {
 #define bad_arg_msg _("ERROR: Bad argument '%s' to %s option.\n")
 #define see_help_msg _("\nRun `gaf export --help' for more information.\n")
 
-/* Main function for `gaf export' */
-void
-cmd_export (int argc, char **argv)
+static void
+cmd_export_impl (void *data, int argc, char **argv)
 {
   int i;
   GError *err = NULL;
@@ -1094,3 +1093,11 @@ export_command_line (int argc, char * const *argv)
     exit (1);
   }
 }
+
+/* Main function for `gaf export' */
+int
+cmd_export (int argc, char **argv)
+{
+  scm_boot_guile (argc, argv, cmd_export_impl, NULL); /* Doesn't return */
+  return 0;
+}
-- 
2.1.4

